From eea72e2a21f622fac09d27cf9bb1b78fcfbbd491 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Mon, 20 Nov 2017 19:23:00 +0200
Subject: [PATCH 2825/5242] MLK-16949: arm64: dts: fsl-imx8mq: Reorganize dcss
 drm nodes

commit  2df225d45522224725608d800610f724c3e0b049 from
https://source.codeaurora.org/external/imx/linux-imx.git

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>

Also add _drm nodes to fsl-imx8mq-evk and remove imx8mq-evk-drm.dts

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    1 -
 .../boot/dts/freescale/fsl-imx8mq-evk-drm.dts      |   94 --------------------
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts   |   25 ++++++
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi      |   49 ++++++++++
 4 files changed, 74 insertions(+), 95 deletions(-)
 delete mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-drm.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 3310e80..dd1150a 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -48,6 +48,5 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-dsi-adv7535.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-dsi-rm67191.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-evk.dtb \
-				 fsl-imx8mq-evk-drm.dtb \
 				 fsl-imx8mq-evk-m4.dtb \
 				 fsl-imx8mq-evk-pcie1-m2.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-drm.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-drm.dts
deleted file mode 100644
index d2df468..0000000
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-drm.dts
+++ /dev/null
@@ -1,94 +0,0 @@
-/*
- * Copyright 2017 NXP
- *
- * This program is free software; you can redistribute it and/or
- * modify it under the terms of the GNU General Public License
- * as published by the Free Software Foundation; either version 2
- * of the License, or (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- */
-
-#include "fsl-imx8mq-evk.dts"
-
-/ {
-	dcss_drm: dcss@0x32e00000 {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		compatible = "nxp,imx8mq-dcss";
-		reg = <0x0 0x32e00000 0x0 0x30000>;
-		interrupts = <3 IRQ_TYPE_LEVEL_HIGH>, <4 IRQ_TYPE_LEVEL_HIGH>, <5 IRQ_TYPE_LEVEL_HIGH>,
-				     <6 IRQ_TYPE_LEVEL_HIGH>, <8 IRQ_TYPE_EDGE_RISING>;
-		interrupt-names = "dpr_dc_ch0", "dpr_dc_ch1", "dpr_dc_ch2", "ctx_ld",
-						  "dtg_prg1";
-		interrupt-parent = <&irqsteer_dcss>;
-		clocks = <&clk IMX8MQ_CLK_DISP_APB_ROOT>,
-			 <&clk IMX8MQ_CLK_DISP_AXI_ROOT>,
-			 <&clk IMX8MQ_CLK_DISP_RTRM_ROOT>,
-			 <&clk IMX8MQ_CLK_DC_PIXEL_DIV>,
-			 <&clk IMX8MQ_CLK_DISP_DTRC_DIV>;
-		clock-names = "apb", "axi", "rtrm", "pixel", "dtrc";
-		assigned-clocks = <&clk IMX8MQ_CLK_DC_PIXEL_SRC>,
-				  <&clk IMX8MQ_CLK_DISP_AXI_SRC>,
-				  <&clk IMX8MQ_CLK_DISP_APB_SRC>,
-				  <&clk IMX8MQ_CLK_DISP_RTRM_SRC>,
-				  <&clk IMX8MQ_CLK_DISP_RTRM_PRE_DIV>;
-		assigned-clock-parents = <&clk IMX8MQ_VIDEO_PLL1_OUT>,
-					 <&clk IMX8MQ_SYS1_PLL_800M>,
-					 <&clk IMX8MQ_SYS2_PLL_125M>,
-					 <&clk IMX8MQ_SYS1_PLL_800M>;
-		assigned-clock-rates = <594000000>,
-				       <800000000>,
-				       <125000000>,
-				       <400000000>,
-				       <400000000>;
-
-		disp-dev = "hdmi_disp";
-
-		status = "okay";
-
-		dcss_disp0: port@0 {
-			reg = <0>;
-
-			dcss_disp0_hdmi: hdmi-endpoint {
-				remote-endpoint = <&hdmi_disp>;
-			};
-		};
-	};
-
-	display-subsystem {
-		compatible = "fsl,imx-display-subsystem";
-		ports = <&dcss_disp0>;
-	};
-
-	hdmi_drm: hdmi_drm@32c00000 {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		compatible = "fsl,imx8mq-hdmi";
-		reg = <0x0 0x32c00000 0x0 0x33800>,	/* HDP registers */
-				<0x0 0x32e40000 0x0 0x40000>;   /* HDP SEC register */
-		interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>;
-		power-domains = <&disp_pd>;
-		port@0 {
-			reg = <0>;
-				hdmi_disp: endpoint {
-				remote-endpoint = <&dcss_disp0_hdmi>;
-			};
-		};
-	};
-};
-
-&hdmi_cec {
-	status = "okay";
-};
-
-&dcss {
-	status = "disabled";
-};
-
-&hdmi {
-	status = "disabled";
-};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index 41bc2e2..7deb8a9 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -656,3 +656,28 @@
 	>;
 	dc-supply = <&reg_gpio_dvfs>;
 };
+
+&dcss_drm {
+	status = "okay";
+
+	disp-dev = "hdmi_disp";
+	dcss_disp0: port@0 {
+		reg = <0>;
+
+		dcss_disp0_hdmi: hdmi-endpoint {
+			remote-endpoint = <&hdmi_disp>;
+		};
+	};
+
+};
+
+&hdmi_drm {
+	status = "okay";
+
+	port@0 {
+		reg = <0>;
+			hdmi_disp: endpoint {
+			remote-endpoint = <&dcss_disp0_hdmi>;
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index fb233f7..96a3803 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -539,6 +539,55 @@
 		status = "disabled";
 	};
 
+	dcss_drm: dcss@0x32e00000 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "nxp,imx8mq-dcss";
+		reg = <0x0 0x32e00000 0x0 0x30000>;
+		interrupts = <3 IRQ_TYPE_LEVEL_HIGH>,
+			     <4 IRQ_TYPE_LEVEL_HIGH>,
+			     <5 IRQ_TYPE_LEVEL_HIGH>,
+			     <6 IRQ_TYPE_LEVEL_HIGH>,
+			     <8 IRQ_TYPE_EDGE_RISING>;
+		interrupt-names = "dpr_dc_ch0",
+				  "dpr_dc_ch1",
+				  "dpr_dc_ch2",
+				  "ctx_ld",
+				  "dtg_prg1";
+		interrupt-parent = <&irqsteer_dcss>;
+		clocks = <&clk IMX8MQ_CLK_DISP_APB_ROOT>,
+			 <&clk IMX8MQ_CLK_DISP_AXI_ROOT>,
+			 <&clk IMX8MQ_CLK_DISP_RTRM_ROOT>,
+			 <&clk IMX8MQ_CLK_DC_PIXEL_DIV>,
+			 <&clk IMX8MQ_CLK_DISP_DTRC_DIV>;
+		clock-names = "apb", "axi", "rtrm", "pixel", "dtrc";
+		assigned-clocks = <&clk IMX8MQ_CLK_DC_PIXEL_SRC>,
+				  <&clk IMX8MQ_CLK_DISP_AXI_SRC>,
+				  <&clk IMX8MQ_CLK_DISP_APB_SRC>,
+				  <&clk IMX8MQ_CLK_DISP_RTRM_SRC>,
+				  <&clk IMX8MQ_CLK_DISP_RTRM_PRE_DIV>;
+		assigned-clock-parents = <&clk IMX8MQ_VIDEO_PLL1_OUT>,
+					 <&clk IMX8MQ_SYS1_PLL_800M>,
+					 <&clk IMX8MQ_SYS2_PLL_125M>,
+					 <&clk IMX8MQ_SYS1_PLL_800M>;
+		assigned-clock-rates = <594000000>,
+				       <800000000>,
+				       <125000000>,
+				       <400000000>,
+				       <400000000>;
+		status = "disabled";
+	};
+
+	hdmi_drm: hdmi_drm@32c00000 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "fsl,imx8mq-hdmi";
+		reg = <0x0 0x32c00000 0x0 0x33800>,	/* HDP registers */
+				<0x0 0x32e40000 0x0 0x40000>;   /* HDP SEC register */
+		interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>;
+		status = "disabled";
+	};
+
 	iomuxc: iomuxc@30330000 {
 		compatible = "fsl,imx8mq-iomuxc";
 		reg = <0x0 0x30330000 0x0 0x10000>;
-- 
1.7.9.5

